File: src/ast/PackedTemplateParameterExpansion.java
package ast;

import java.io.StringWriter;

public class PackedTemplateParameterExpansion extends ParentNode {
    public PackedTemplateParameterExpansion(BaseNode child) {
        super(NodeType.PackedTemplateParameterExpansion, child);
    }

    @Override
    public void PrintLeft(StringWriter writer) {
        if (child instanceof PackedTemplateParameter) {
            if (((PackedTemplateParameter) child).Nodes.size() != 0) {
                child.Print(writer);
            }
        } else {
            writer.write("...");
        }
    }
}

File: src/de/fearlesstobi/demangler/ast/PackedTemplateParameterExpansion.kt
package de.fearlesstobi.demangler.ast

import java.io.StringWriter

class PackedTemplateParameterExpansion(child: BaseNode?) : ParentNode(NodeType.PackedTemplateParameterExpansion, child) {
    public override fun printLeft(writer: StringWriter) {
        if (child is PackedTemplateParameter) {
            if (!child.nodes!!.isEmpty()) {
                child.print(writer)
            }
        } else {
            writer.write("...")
        }
    }
}